firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -168,8 +183,12 @@ static FirebaseFirestore newInstance(
     this.firebaseApp = firebaseApp;
     this.instanceRegistry = instanceRegistry;
     this.metadataProvider = metadataProvider;
+    this.emulatorSettings = emulatorSettings;
 
-    settings = new FirebaseFirestoreSettings.Builder().build();
+    this.settings = new FirebaseFirestoreSettings.Builder().build();
+    if (this.emulatorSettings != null) {
+      this.settings = mergeEmulatorSettings(settings, emulatorSettings);
```

Comment:
Could mergeEmulatorSettings take a Nullable `emulatorSettings`? That would remove the if check here and once more below.